### PR TITLE
fix namespace constructor in computed constructor

### PR DESCRIPTION
**Pull request by shabanovd**
close https://github.com/eXist-db/exist/issues/1237

### CODE REVIEW

File: src/org/exist/dom/memtree/MemTreeBuilder.java [hunk 1]
```diff
@@ -415,12 +416,26 @@ public int namespaceNode(final String prefix, final String uri) {
...
+    public int namespaceNode(final QName qname, boolean checkNS) {
         final int lastNode = doc.getLastNode();
         boolean addNode = true;
         if(doc.nodeName != null) {
             final QName elemQN = doc.nodeName[lastNode];
             if(elemQN != null) {
                 final String elemPrefix = (elemQN.getPrefix() == null) ? XMLConstants.DEFAULT_NS_PREFIX : elemQN.getPrefix();
+                if (checkNS
+                    && XMLConstants.DEFAULT_NS_PREFIX.equals(elemPrefix)
+                    && "xmlns".equals(qname.getPrefix())
```
Comment (adamretter) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: Consider using `XMLConstants.XMLNS_ATTRIBUTE`

File: src/org/exist/dom/memtree/MemTreeBuilder.java [hunk 2]
```diff
@@ -415,12 +416,26 @@ public int namespaceNode(final String prefix, final String uri) {
...
+                    && "".equals(qname.getLocalPart())) {
```
Comment (adamretter) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: Should this not be:
```
&&  XMLConstants.DEFAULT_NS_PREFIX.equals(qname.getNamespaceURI())
```

File: src/org/exist/dom/memtree/MemTreeBuilder.java (hunk 1 above)
Comment (shabanovd) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: done

File: src/org/exist/dom/memtree/MemTreeBuilder.java (hunk 2 above)
Comment (shabanovd) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: nope, why it should check that `default namespace prefix` is equal to namespace uri?

Comment (adamretter) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: Sorry I made a copy-paste error. Should be:

```java
&&  XMLConstants.NULL_NS_URI.equals(qname.getNamespaceURI())
```
As checking the local part of a QName used for the namespace doesn't make any sense.

Comment (dizzzz) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: subtile :-)

Comment (shabanovd) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: @adamretter I think you didn't understand what and how was fixed here. I have no idea why you want to check namespace-URI to be "" (empty string) if original issue was in replacing default namespace with **any** namespace-URI

Comment (dizzzz) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: @adamretter do you still have a concern?

Comment (dizzzz) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: technically there is no difference.... it is the empty string ""

Comment (dizzzz) [inline src/org/exist/dom/memtree/MemTreeBuilder.java]: probably I just dont see it :-)

**Pull request merged**